File: src/main/java/nl/joelchrist/shorty/shorties/repositories/ShortiesRepository.java
package nl.joelchrist.shorty.shorties.repositories;

import nl.joelchrist.shorty.shorties.domain.Shorty;
import org.jongo.MongoCollection;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public class ShortiesRepository {

    @Autowired
    private MongoCollection shortiesCollection;

    public void saveShorty(Shorty shorty) {
        shortiesCollection.save(shorty);
    }

    public Optional<Shorty> findShorty(String shortIdentifier) {
        Shorty shorty = shortiesCollection.findOne("{shortIdentifier: #}", shortIdentifier).as(Shorty.class);
        return Optional.ofNullable(shorty);
    }
}

File: src/main/kotlin/nl/joelchrist/shorty/shorties/repositories/ShortiesRepository.kt
package nl.joelchrist.shorty.shorties.repositories

import nl.joelchrist.shorty.shorties.domain.Shorty
import org.jongo.MongoCollection
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Repository

@Repository
class ShortiesRepository(@Autowired private val shortiesCollection: MongoCollection) {
    fun save(entity: Shorty) {
        shortiesCollection.save(entity)
    }

    fun findByIdentifier(identifier: String) : Shorty? = shortiesCollection.findOne("{identifier: '$identifier'}").`as`(Shorty::class.java)

    fun findByOwner(owner: String) : List<Shorty>? = shortiesCollection.find("{owner: '$owner'}").`as`(Shorty::class.java).toList()

    fun findAll(): List<Shorty>? = shortiesCollection.find().`as`(Shorty::class.java).toList()
}